security-service-client-spring/src/main/java/org/zowe/apiml/security/client/service/GatewaySecurityService.java

```diff
@@ -89,26 +98,27 @@ public QueryResponse query(String token) {
             gatewayConfigProperties.getHostname(), authConfigurationProperties.getGatewayQueryEndpoint());
         String cookie = String.format("%s=%s", authConfigurationProperties.getCookieProperties().getCookieName(), token);
 
-        HttpHeaders headers = new HttpHeaders();
-        headers.add(HttpHeaders.COOKIE, cookie);
 
         try {
-            ResponseEntity<QueryResponse> response = restTemplate.exchange(
-                uri,
-                HttpMethod.GET,
-                new HttpEntity<>(headers),
-                QueryResponse.class);
-
-            return response.getBody();
-        } catch (HttpClientErrorException | ResourceAccessException | HttpServerErrorException e) {
-            responseHandler.handleBadResponse(e, ErrorType.TOKEN_NOT_VALID,
-                "Can not access Gateway service. Uri '{}' returned: {}", uri, e.getMessage());
+            HttpGet get = new HttpGet(uri);
+            get.addHeader(HttpHeaders.COOKIE, cookie);
+            CloseableHttpResponse response = closeableHttpClient.execute(get);
+            String responseBody = new BufferedReader(new InputStreamReader(response.getEntity().getContent())).lines().collect(Collectors.joining("\n"));
+            if (response.getStatusLine().getStatusCode() > 299) {
+                ErrorType errorType = getErrorType(responseBody);
+                responseHandler.handleErrorType(response, errorType,
+                    "Cannot access Gateway service. Uri '{}' returned: {}", uri);
+                return null;
+            }
+            ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
may be constant

Reply:
Resolved by moving the objectMapper to final class variable.